astropy/utils/iers/iers.py

```diff
@@ -679,20 +709,28 @@ def open(cls):
             if cls.iers_table.meta.get('data_url') in all_urls:
                 return cls.iers_table
 
-        try:
-            filename = download_file(all_urls[0], sources=all_urls, cache=True)
-        except Exception as err:
+        for url in all_urls:
```

Comment:
Why the loop? I thought the idea of `sources` in `dowload_url` is exactly to give different possible sources, but still count them as "the same file" for download purposes. Or is this to guard against a mirror providing a bad file? If it is not essential to the PR, I'd recommend not changing it here.

Reply:
Probably needs an upstream fix but that was out of scope. If *any* of the provided URLs are not valid URLs then `download_file` raises an exception, even if one of the URL's can download successfully.

Reply:
That's bad and would seem a real bug in `download_file`. But maybe best not to try to work around it here?

Reply:
Actually the other reason (maybe more important) is that part of the point here is to not only ensure that some HTML page content got downloaded (which is all that `download_file` cares about), but that it is *correct*. That was one of the original problems with a site returning some text that was not the expected IERS table content and crashing later.

In short, we really do need a loop that tries each URL in turn, both downloading and validating the content.

Reply:
OK, let's keep this as is then!

Reply:
If you want to pursue the `download_file` bug, we need to ping @aarchiba on the original intent of that implementation. FYI.